pkg/vm/vm.go

```diff
@@ -109,13 +130,24 @@ func NewWithTrigger(t trigger.Type) *VM {
 	vm := &VM{
 		state:   vmstate.None,
 		trigger: t,
+		hooks:   defaultHooks,
```

Comment:
Please, see the https://github.com/nspcc-dev/neo-go/pull/3462#discussion_r1629453637.

Reply:
Could you please clarify fyfyrchik's point for me?

Reply:
We need to remove an empty `defaultHooks` variable and simply make `hooks` to be nil in VM constructor. Then during hooks execution you can check if `vm.hooks` field is nil and execute it only if it's not nil.

Reply:
This would add one additional indirect jump to the VM's hot execution loop and can cost us performance. That's why I've implemented it using default hook values.

Of course, these discussions are meaningless without performance benchmarks and we probably already have enough indirect jumps in the execution loop so this one won't make a difference.

Should I reimplement it using `!= nil` checks?

Reply:
I really appreciate avoiding branches, but this one doesn't change the entire vm.VM lifecycle, so branch predictor isn't likely to have any problems with it. You can try some benchmarks we have already like `BenchmarkScriptFibonacci`, I believe there won't be any measurable difference and if that's the case then simpler code is preferred.

Reply:
> or are there any that I've missed?

@Furetur, let's also fix this one.

Reply:
If I wasn't mistaken @roman-khimov was against this change.

@roman-khimov, can you please clarify whether I should rewrite the current implementation with default hooks or not?

Reply:
`!= nil` is a simpler thing here, less code, same effect.

Reply:
Done